### PR TITLE
Add some missing things

### PR DESCRIPTION
**Pull request by auscompgeek**

### CODE REVIEW

**Comment by auscompgeek** [comment]
Ah right, the Pose2d specialisation of TimeInterpolatableBuffer would require the Pose2d binding to come first for its function signature.

This is kinda awkward. TimeInterpolatableBuffer doesn't belong in geometry wholesale, but it'd be odd for the root to depend on wpimath.geometry. Hmm.

**Comment by auscompgeek** [comment]
Hmm, I could just have two copies I guess…

**Comment by auscompgeek** [comment]
... or maybe it should just go into its own extension module.

File: docs/conf.py [hunk 1]
```diff
@@ -146,6 +146,7 @@
 gen_package(root, "wpimath.estimator")
 gen_package(root, "wpimath.filter")
 gen_package(root, "wpimath.geometry")
+gen_package(root, "wpimath.interpolation")
```
Comment (virtuald) [inline docs/conf.py:149]: Forgot to include this in api.rst; everything else looks fine, merge at will once this is green

**Pull request merged**